src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -0,0 +1,3138 @@
+
+				numericDriver.displayPopup(HAVE_OLED ? "Interpolation Off" : "OFF");
+			}
+			setDisplayParameterNameTimer();
+		}
+	}
+
+	//when you press affect entire in a kit, the parameter selection needs to reset
+	else if (b == AFFECT_ENTIRE) {
+		initParameterSelection();
+		goto passToOthers;
+	}
+
+	else {
+passToOthers:
+
+		uiNeedsRendering(this);
+
+		ActionResult result = InstrumentClipMinder::buttonAction(b, on, inCardRoutine);
+		if (result != ActionResult::NOT_DEALT_WITH) {
+			return result;
+		}
+
+		result = ClipView::buttonAction(b, on, inCardRoutine);
+
+		setDisplayParameterNameTimer();
+
+		return result;
+	}
+
+	if (on && (b == KEYBOARD || b == CLIP_VIEW || b == SESSION_VIEW)) {
+		numericDriver.cancelPopup();
+	}
+
+	if (on && (b != KEYBOARD && b != CLIP_VIEW && b != SESSION_VIEW)) {
+		setDisplayParameterNameTimer();
+		uiNeedsRendering(this);
+	}
+
+	return ActionResult::DEALT_WITH;
+}
+
+//simplified version of the InstrumentClipView::enterScaleMode function. No need to render any animation.
+void AutomationInstrumentClipView::enterScaleMode(uint8_t yDisplay) {
+
+	char modelStackMemory[MODEL_STACK_MAX_SIZE];
+	ModelStackWithTimelineCounter* modelStack = currentSong->setupModelStackWithCurrentClip(modelStackMemory);
+	InstrumentClip* clip = (InstrumentClip*)modelStack->getTimelineCounter();
+
+	int32_t newRootNote;
+	if (yDisplay == 255) {
+		newRootNote = 2147483647;
+	}
+	else {
+		newRootNote = clip->getYNoteFromYDisplay(yDisplay, currentSong);
+	}
+
+	int32_t newScroll = instrumentClipView.setupForEnteringScaleMode(newRootNote, yDisplay);
+
+	// See which NoteRows need to animate
+	PadLEDs::numAnimatedRows = 0;
+	for (int32_t i = 0; i < clip->noteRows.getNumElements(); i++) {
+		NoteRow* thisNoteRow = clip->noteRows.getElement(i);
+		int32_t yVisualTo = clip->getYVisualFromYNote(thisNoteRow->y, currentSong);
+		int32_t yDisplayTo = yVisualTo - newScroll;
+		int32_t yDisplayFrom = thisNoteRow->y - clip->yScroll;
+
+		// If this NoteRow is going to end up on-screen or come from on-screen...
+		if ((yDisplayTo >= 0 && yDisplayTo < kDisplayHeight) || (yDisplayFrom >= 0 && yDisplayFrom < kDisplayHeight)) {
+
+			ModelStackWithNoteRow* modelStackWithNoteRow = modelStack->addNoteRow(thisNoteRow->y, thisNoteRow);
+
+			PadLEDs::animatedRowGoingTo[PadLEDs::numAnimatedRows] = yDisplayTo;
+			PadLEDs::animatedRowGoingFrom[PadLEDs::numAnimatedRows] = yDisplayFrom;
+			uint8_t mainColour[3];
+			uint8_t tailColour[3];
+			uint8_t blurColour[3];
+			clip->getMainColourFromY(thisNoteRow->y, thisNoteRow->getColourOffset(clip), mainColour);
+			getTailColour(tailColour, mainColour);
+			getBlurColour(blurColour, mainColour);
+
+			instrumentClipView.drawMuteSquare(thisNoteRow, PadLEDs::imageStore[PadLEDs::numAnimatedRows],
+			                                  PadLEDs::occupancyMaskStore[PadLEDs::numAnimatedRows]);
+			PadLEDs::numAnimatedRows++;
+			if (PadLEDs::numAnimatedRows >= kMaxNumAnimatedRows) {
+				break;
+			}
```

Comment:
this code may not be required since we're not rendering the note data at all in the automation view?